DSharpPlus/Entities/Interaction/DiscordHttpInteraction.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using DSharpPlus.Net.Abstractions;
+using DSharpPlus.Net.Serialization;
+using Newtonsoft.Json;
+
+namespace DSharpPlus.Entities;
+
+public class DiscordHttpInteraction : DiscordInteraction
+{
+    [JsonIgnore]
+    internal readonly TaskCompletionSource taskCompletionSource = new();
+    
+    [JsonIgnore]
+    internal byte[] response;
+
+    internal bool Cancel() => this.taskCompletionSource.TrySetCanceled();
+
+    internal async Task<byte[]> GetResponseAsync()
+    {
+        await this.taskCompletionSource.Task;
+
+        return this.response;
+    }
+    
+    public override Task CreateResponseAsync(DiscordInteractionResponseType type, DiscordInteractionResponseBuilder? builder = null)
+    {
+        if (this.taskCompletionSource.Task.IsCanceled)
+        {
+            throw new InvalidOperationException(
```

Comment:
This isn't necessarily indicative of Discord closing the connection though?

Reply:
The cancelation token should be tied to the http connection. Maybe i should add additional docs, in aspnet you get a ct which is cancled when the request is cancled

Reply:
I think velvet means that the http connection could also be closed due to other reasons, such as a random network drop?

Reply:
> "This is likely"
I think its the common case.